// gitbook-plugin-onedata-screencast-theme/index.js
var marked = require('marked');

module.exports = {
    website: {
        assets: "./book",
        css: [
          "onedata.css"
        ],
        html: {
            "html:start": function() {
                return "<!-- Start book "+this.options.title+" -->";
            },
            "html:end": function() {
                return "<!-- End of book "+this.options.title+" -->";
            },

            "head:start": "<!-- head:start -->",
            "head:end": "<!-- head:end -->",

            "body:start": "<!-- body:start -->",
            "body:end": "<!-- body:end -->"
        },
    },
    blocks: {
      slide: {
        blocks: ['onscreen', 'script'],
        process: function(blk) {
          var title = blk.kwargs.title;
          var html = '<div class="k-slide"><div class="title">'+marked(title)+'</div>';
          blk.blocks.forEach(function(block) {
            var divClass = block.name;
            var content;
            if (divClass === 'script') {
              content = '<pre>' + marked(block.body) + '</pre>';
            } else {
              content = marked(block.body);
            }
            html += '<div class="k-' + divClass + '">' + content + '</div>';
          });
          html += '</div>';
          return html;
        },
      },
    }
};
